### ISSUE TITLE
Compilation failure on MacOS Mojave

### ISSUE THREAD

**Issue by antanst** [tevador/RandomX#30]
MacOS Mojave, Apple LLVM version 10.0.0

Compilation error on both `master@6e8c83fd` and `dev@6b344b81`:

```
$ make
cc -march=native -O3 -flto -c src/argon2_core.c -o obj/argon2_core.o
cc -march=native -O3 -flto -c src/argon2_ref.c -o obj/argon2_ref.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/AssemblyGeneratorX86.cpp -o obj/AssemblyGeneratorX86.o
cc -march=native -O3 -flto -c src/blake2/blake2b.c -o obj/blake2b.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/CompiledVirtualMachine.cpp -o obj/CompiledVirtualMachine.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/dataset.cpp -o obj/dataset.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/JitCompilerX86.cpp -o obj/JitCompilerX86.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/instructionsPortable.cpp -o obj/instructionsPortable.o
src/instructionsPortable.cpp:22:26: warning: expected 'ON' or 'OFF' or 'DEFAULT' in pragma [-Wunknown-pragmas]
#pragma STDC FENV_ACCESS on
                         ^
1 warning generated.
c++ -std=c++11 -maes -march=native -O3 -flto -c src/Instruction.cpp -o obj/Instruction.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/InterpretedVirtualMachine.cpp -o obj/InterpretedVirtualMachine.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/main.cpp -o obj/main.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/Program.cpp -o obj/Program.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/softAes.cpp -o obj/softAes.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/VirtualMachine.cpp -o obj/VirtualMachine.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/Cache.cpp -o obj/Cache.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/virtualMemory.cpp -o obj/virtualMemory.o
cc -march=native -O3 -flto -c src/reciprocal.c -o obj/reciprocal.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/LightClientAsyncWorker.cpp -o obj/LightClientAsyncWorker.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/hashAes1Rx4.cpp -o obj/hashAes1Rx4.o
c++ -x assembler-with-cpp -c src/JitCompilerX86-static.S -o obj/JitCompilerX86-static.o
src/JitCompilerX86-static.S:41:8: error: invalid alignment value
.align 64
       ^
src/JitCompilerX86-static.S:45:8: error: invalid alignment value
.align 64
       ^
src/JitCompilerX86-static.S:48:8: error: invalid alignment value
.align 64
       ^
src/JitCompilerX86-static.S:67:8: error: invalid alignment value
.align 64
       ^
src/JitCompilerX86-static.S:71:8: error: invalid alignment value
.align 64
       ^
make: *** [obj/JitCompilerX86-static.o] Error 1
```

**Comment by tevador** [comment]
Can you try 1945aae9f86cef669517ab60e8725cc76c78592f (dev branch)?

**Comment by antanst** [comment]
Fails with a different error now:

```
$ make
cc -march=native -O3 -flto -c src/argon2_core.c -o obj/argon2_core.o
cc -march=native -O3 -flto -c src/argon2_ref.c -o obj/argon2_ref.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/AssemblyGeneratorX86.cpp -o obj/AssemblyGeneratorX86.o
cc -march=native -O3 -flto -c src/blake2/blake2b.c -o obj/blake2b.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/CompiledVirtualMachine.cpp -o obj/CompiledVirtualMachine.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/dataset.cpp -o obj/dataset.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/JitCompilerX86.cpp -o obj/JitCompilerX86.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/instructionsPortable.cpp -o obj/instructionsPortable.o
src/instructionsPortable.cpp:22:26: warning: expected 'ON' or 'OFF' or 'DEFAULT' in pragma [-Wunknown-pragmas]
#pragma STDC FENV_ACCESS on
                         ^
1 warning generated.
c++ -std=c++11 -maes -march=native -O3 -flto -c src/Instruction.cpp -o obj/Instruction.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/InterpretedVirtualMachine.cpp -o obj/InterpretedVirtualMachine.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/main.cpp -o obj/main.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/Program.cpp -o obj/Program.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/softAes.cpp -o obj/softAes.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/VirtualMachine.cpp -o obj/VirtualMachine.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/Cache.cpp -o obj/Cache.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/virtualMemory.cpp -o obj/virtualMemory.o
cc -march=native -O3 -flto -c src/reciprocal.c -o obj/reciprocal.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/LightClientAsyncWorker.cpp -o obj/LightClientAsyncWorker.o
c++ -std=c++11 -maes -march=native -O3 -flto -c src/hashAes1Rx4.cpp -o obj/hashAes1Rx4.o
c++ -x assembler-with-cpp -c src/JitCompilerX86-static.S -o obj/JitCompilerX86-static.o
c++ -x assembler-with-cpp -c src/squareHash.S -o obj/squareHash.o
c++ obj/argon2_core.o obj/argon2_ref.o obj/AssemblyGeneratorX86.o obj/blake2b.o obj/CompiledVirtualMachine.o obj/dataset.o obj/JitCompilerX86.o obj/instructionsPortable.o obj/Instruction.o obj/InterpretedVirtualM
achine.o obj/main.o obj/Program.o obj/softAes.o obj/VirtualMachine.o obj/Cache.o obj/virtualMemory.o obj/reciprocal.o obj/LightClientAsyncWorker.o obj/hashAes1Rx4.o obj/JitCompilerX86-static.o obj/squareHash.o -l
pthread -o bin/randomx
undef: squareHash
Undefined symbols for architecture x86_64:
  "squareHash", referenced from:
      initBlock_loop in squareHash.o
     (maybe you meant: _squareHash)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [bin/randomx] Error 1
```

**Comment by tevador** [comment]
Try with this fix: 1617d8e34e395f8c22b5beace57a6f8142a02638

**Comment by antanst** [comment]
As of `1617d8e` it compiles just fine. 👍
